Objective-C/CBLDatabase.h

```diff
@@ -142,31 +143,42 @@ typedef struct {
 /** Compiles a Couchbase Lite query, from any of several input formats, specifying sorting.
     Once compiled, the query can be run many times with different parameter values.
     @param where  The query specification; see above for details.
-    @param sortDescriptors  An array of NSSortDescriptors specifying how to sort the result.
+                    (Corresponds to the WHERE clause of a SQL or N1QL query.)
+    @param sortDescriptors  An array of NSSortDescriptors or NSStrings, specifying how to sort the
+                    resulting rows. These can name key-paths or be NSExpresson format strings.
+                    If nil, no sorting occurs; this is faster but the order of rows is undefined.
+                    (Corresponds to the ORDER BY clause of a SQL or N1QL query.)
+    @param returning  An array of NSExpressions (or expression format strings) describing values
+                    to include in the result. 
+                    If nil, only the document ID and sequence number will be available.
+                    (Corresponds to the SELECT clause of a SQL or N1QL query.)
     @param error  If the query cannot be parsed, an error will be stored here.
     @return  The CBLQuery, or nil on error. */
 - (nullable CBLQuery*) createQueryWhere: (nullable id)where
                                 orderBy: (nullable NSArray*)sortDescriptors
+                              returning: (nullable NSArray*)returning
                                   error: (NSError**)error;
 
 /** Creates a value index (type kCBLValueIndex) on a given document property.
     This will speed up queries that test that property, at the expense of making database writes a
     little bit slower.
-    @param expressions  Expressions to index, typically key-paths.
+    @param expressions  Expressions to index, typically key-paths. Can be NSExpression objects,
+                    or NSStrings that are expression format strings.
     @param error  If an error occurs, it will be stored here if this parameter is non-NULL.
     @return  True on success, false on failure. */
-- (bool) createIndexOn: (NSArray<NSExpression*>*)expressions
+- (bool) createIndexOn: (NSArray*)expressions
```

Comment:
Yes. I think we always use BOOL in 1.0 API and Apple Objective-C Foundation mostly uses BOOL.